### ISSUE TITLE
Upgrading to 5.1.0 causes module not found errors with Sage 10.0.0-beta.2

### ISSUE THREAD

**Issue by cspicuzza** [roots/bud#1006]
### Agreement

- [X] This is not a duplicate of an existing issue
- [X] I have read the [guidelines for Contributing to Roots Projects](https://github.com/roots/.github/blob/master/CONTRIBUTING.md)
- [X] This is not a personal support request that should be posted on the [Roots Discourse](https://discourse.roots.io/) community

### Expected Behavior

I was catching up on the [latest dialog](https://discourse.roots.io/t/sage-10-0-0-beta-2-released/21923/39) in the Roots discourse for Sage 10-beta2 before attempting to try the theme out. 

- I pulled down the latest sage build
- ran `composer install`
- ran `yarn install`
- ran `yarn dev`

As expected, the build completed

I then bumped the Bud devDependency version to 5.1.0. I expected to be able to rebuild without issues

### Actual Behavior

Builds fail with the following:
```
✘ Module not found: Error: Can't resolve '@scripts/components' in './resources/scripts'
✘ Module not found: Error: Can't resolve '@wordpress/edit-post' in './resources/scripts'
✘ Module not found: Error: Can't resolve '@wordpress/blocks' in './resources/scripts'
✘ Module not found: Error: Can't resolve '@scripts/components' in './resources/scripts'
```

### Steps To Reproduce

1. Do a fresh install of sage 10.0.0-beta.2
2. change the Bud dependency: `"@roots/bud": "^5.1.0",`
3. run `composer install`
4. run `yarn install`
5. run `yarn dev`

### version

@roots/bud/5.1.0 darwin-arm64 node-v16.13.2

### What package manager are you using?

yarn classic

### version

1.22.17

### Are you using pnpm?

- [ ] yes
- [X] no

### Logs

```shell
yarn run v1.22.17
$ bud dev
✘ Module not found: Error: Can't resolve '@scripts/components' in './resources/scripts'
✘ Module not found: Error: Can't resolve '@wordpress/edit-post' in './resources/scripts'
✘ Module not found: Error: Can't resolve '@wordpress/blocks' in './resources/scripts'
✘ Module not found: Error: Can't resolve '@scripts/components' in './resources/scripts'
```
```


### Configuration

```shell
defaults from https://github.com/roots/sage/tree/10.0.0-beta.2
```

**Comment by kellymears** [comment]
Thanks for trying it out. It's probably easier to use sage's main branch as a starting point since there are a number of changes that need to be made in order to use 5.1.0. 

[Check out the section of the release notes dedicated to @roots/sage](https://github.com/roots/bud/releases/tag/v5.1.0). 

## Update dependencies

You'll need to do more than just bump `@roots/bud` to `5.1.0`. All of the bud packages have the same version, so anything in the `@roots` organization namespace should  be `5.1.0`. 

You'll also need to explicitly install a several extensions that were previously included in `@roots/sage` but are now left for the theme to opt-in to. This is to make them easier to remove should a user not need or want them. 

Dependencies should look like this:

https://github.com/roots/sage/blob/f63033ef3ae9312ea27d07d3be075cbd0ccff8ff/package.json#L24-L43

If you were to think that this is not the most sustainable way to handle aligning dependencies, I agree. I anticipate this will pass review and be included in a 5.2.0 update in the very immediate future:

- https://github.com/roots/bud/pull/1005

## update bud.config.js

The `bud.assets` call requires an absolute path. There is a helper function for this:

https://github.com/roots/sage/blob/f63033ef3ae9312ea27d07d3be075cbd0ccff8ff/bud.config.js#L19-L23

You must specify a publicPath:

https://github.com/roots/sage/blob/f63033ef3ae9312ea27d07d3be075cbd0ccff8ff/bud.config.js#L42-L45

## bump acorn

You'll need to update `roots/acorn` for it to be able to parse the entrypoints:

https://github.com/roots/sage/blob/f63033ef3ae9312ea27d07d3be075cbd0ccff8ff/composer.json#L43

**Comment by cspicuzza** [comment]
@kellymears I appreciate the response. I did exactly that and switched to the main branch for now and was able to build the project. I'll close this issue for now.
